modules/polygon/src/polygon-utils.d.ts

```diff
@@ -1,85 +1,120 @@
+ * Checks winding direction of the polygon and reverses the polygon in case of opposite winding direction.
  * Note: points are modified in-place.
- * @param points An array that represents points of the polygon.
- * @param start Start index of the polygon in the points array.
- * @param end End index of the polygon in the points array.
- * @param size Size of a point, 2 (XZ) or 3 (XYZ).
+ * @param points Array of points that represent the polygon.
  * @param direction Requested winding direction. 1 is for clockwise, -1 for counterclockwise winding direction.
+ * @param options Parameters of the polygon.
+ * @return Returns true if the winding direction was changed.
  */
-export function modifyPolygonWindingDirectionFlat(points: FlatPointsArray, start: number, end: number, size: number, direction: number): void;
+export function modifyPolygonWindingDirectionPoints(
+  points: PointsArray,
+  direction: number,
+  options?: PolygonParams
+): boolean;
 
 /**
  * Returns winding direction of the polygon.
- * @param points An array that represents points of the polygon.
- * @param start Start index of the polygon in the points array.
- * @param end End index of the polygon in the points array.
- * @param size Size of a point, 2 (XZ) or 3 (XYZ).
+ * @param points Array of points that represent the polygon.
+ * @param options Parameters of the polygon.
  * @returns Winding direction of the polygon.
  */
-export function getPolygonWindingDirectionFlat(points: FlatPointsArray, start: number, end: number, size: number): number;
+export function getPolygonWindingDirectionPoints(
+  points: PointsArray,
+  options?: PolygonParams
+): number;
 
 /**
  * Returns signed area of the polygon.
- * @param points An array that represents points of the polygon.
- * @param start Start index of the polygon in the points array.
- * @param end End index of the polygon in the points array.
- * @param size Size of a point, 2 (XZ) or 3 (XYZ).
+ * @param points Array of points that represent the polygon.
+ * @param options Parameters of the polygon.
  * @returns Signed area of the polygon.
  */
-export function getPolygonSignedAreaFlat(points: FlatPointsArray, start: number, end: number, size: number): number;
+export function getPolygonSignedAreaPoints(points: PointsArray, options?: PolygonParams): number;
 
 /**
  * Calls visitor callback for each segment in the polygon.
- * @param points An array that represents points of the polygon
- * @param start Start index of the polygon in the points array.
- * @param end End index of the polygon in the points array.
- * @param size Size of a point, 2 (XZ) or 3 (XYZ).
+ * @param points Array of points that represent the polygon.
  * @param visitor A callback to call for each segment.
+ * @param options Parameters of the polygon.
  */
-export function forEachSegmentInPolygonFlat(points: FlatPointsArray, start: number, end: number, size: number, visitor: SegmentVisitorFlat): void;
-
-
+export function forEachSegmentInPolygonPoints(
```

Comment:
Are we renaming the `Point` functions? Is this a breaking change?

Reply:
Probably you look on the diff between two commits. Both *- flat and *- points utility functions are new. I just removed flat suffix as it is going to be default one, and added * points suffix to functions used to provide backwards compatibily for the polygon class, but not exported out of the module.